packages/react/menu/src/Menu.tsx

```diff
@@ -601,7 +604,9 @@ const MenuItem = React.forwardRef((props, forwardedRef) => {
       onMouseUp={composeEventHandlers(props.onMouseUp, handleSelect)}
       onMouseLeave={composeEventHandlers(props.onMouseLeave, () => contentContext.onItemLeave())}
       onKeyDown={composeEventHandlers(props.onKeyDown, (event) => {
-        if (!disabled && SELECTION_KEYS.includes(event.key)) {
+        const isTypingAhead = contentContext.searchRef.current !== '' && event.key === ' ';
+        if (disabled || isTypingAhead) return;
```

Comment:
I still find it odd to read this, this would make more sense to me to read:

```suggestion
        const isTypingAhead = contentContext.searchRef.current !== '';
        if (disabled || (isTypingAhead && event.key === ' ')) return;
```

What do you think?

Reply:
Agreed